src/main/java/edu/ucla/library/iiif/auth/delegate/hauth/HauthItem.java

```diff
@@ -0,0 +1,120 @@
+import info.freelibrary.util.LoggerFactory;
+import info.freelibrary.util.StringUtils;
+
+import edu.ucla.library.iiif.auth.delegate.MessageCodes;
+
+/**
+ * A client for interacting with the Hauth service.
+ */
+public class HauthItem {
+
+    /**
+     * The Hauth item logger.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(HauthItem.class, MessageCodes.BUNDLE);
+
+    /**
+     * An internal HTTP client.
+     */
+    private static final HttpClient HTTP = HttpClient.newHttpClient();
+
+    /**
+     * An object mapper for reading JSON.
+     */
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    /**
+     * The access URL of the Hauth service.
+     */
+    private final String myAccessService;
+
+    /**
+     * The ID of the item in question.
+     */
+    private final String myID;
+
+    /**
+     * Creates a new Hauth client.
+     *
+     * @param aService The URL of the authorization access service
+     * @param aID The ID of the item
+     */
+    public HauthItem(final String aService, final String aID) {
+        myAccessService = aService;
+        myID = aID;
+    }
+
+    /**
+     * Gets the ID of this item.
+     *
+     * @return The ID of this item
+     */
+    public String getID() {
+        return myID;
+    }
+
+    /**
+     * Returns whether access to the object with the supplied ID is restricted.
+     *
+     * @return Whether access to the object with the supplied ID is restricted
+     */
+    public boolean isRestricted() throws IOException {
+        final URI uri = URI.create(StringUtils.format(myAccessService, myID));
+        final HttpRequest request = HttpRequest.newBuilder().uri(uri).build();
+
+        LOGGER.debug(MessageCodes.CAD_005, request.method(), uri);
+
+        try {
+            final HttpResponse<String> response = HTTP.send(request, BodyHandlers.ofString());
+
+            switch (response.statusCode()) {
+                case 200:
+                    final String authResponseJSON = response.body();
+                    final JsonNode node = MAPPER.readTree(authResponseJSON);
+                    final List<String> ids = node.findValuesAsText(HauthResponse.ID_KEY);
+                    final List<String> accessCodes = node.findValuesAsText(HauthResponse.ACCESS_KEY);
+
+                    if (!ids.isEmpty() && !accessCodes.isEmpty()) {
+                        final String accessCode = accessCodes.get(0);
+                        final String id = ids.get(0);
+
+                        if (myID.equals(id)) {
+                            LOGGER.debug(MessageCodes.CAD_010, myID, accessCode);
+                            return Boolean.valueOf(accessCode);
+                        }
+
+                        LOGGER.error(MessageCodes.CAD_007, id, myID);
+                    } else {
+                        LOGGER.error(MessageCodes.CAD_008, authResponseJSON);
+                    }
+
+                    break;
```

Comment:
Since [for HTTP 200 responses, the implementation unconditionally puts a String `id` (the same one sent in the request) and a boolean `restricted` in the response body](https://github.com/UCLALibrary/hauth/blob/20af14a3792c600c886e73f8a29d318fa2fd409b/src/main/java/edu/ucla/library/iiif/auth/handlers/AccessLevelHandler.java#L54-L59), is this validation necessary? Regardless of the behavior of the database service underlying that request handler, reaching L101 or L103 is impossible. (Although I think the Hauth OpenAPI contract should probably be explicit that the ID returned is the same as the ID passed in.)

Reply:
I believe returning the ID was my idea. I didn't have a specific use case in mind for including it, so I am fine with removing it. Although, I do think HTTP 200 is the correct code in both cases, and I think JSON is better than a plaintext response, even if it's just one value, since the application/json content type provides some explicit type information about the data (and it's cheap, approx. 20 extra bytes in the response compared with plaintext).

Reply:
my idea / my mistake